clio-client/src/main/scala/org/broadinstitute/clio/client/util/IoUtil.scala

```diff
@@ -8,8 +8,11 @@ import java.util.Comparator
 import scala.sys.process.Process
 
 trait IoUtil {
+
   val googleCloudStorageScheme = "gs"
 
+  protected def gsUtil: IoUtil.GsUtil
```

Comment:
The way `IoUtil` and `GsUtil` work together it's not clear to me why they're different objects. Especially since IoUtil contains google APIs.

Reply:
I think it's mainly a byproduct of a past design, but right now the separation is useful for testing. We can inject a mock `GsUtil` into an `IoUtil` to check that the `IoUtil` code does the right thing assuming some stdout from the external `gsutil` call. If we moved the `Process(Seq("gsutil", ...)).!` calls into `IoUtil`, mocking them would be more complicated.

Reply:
Maybe better names would help. Right now it sounds like they're both util classes, one of which does 'IO' and the other goes 'GS' operations. But in fact `IoUtil` includes google APIs, and `GsUtil` is the internal layer that mediates between public google APIs and the gsutil command line program.